presidio-analyzer/tests/test_spacy_recognizer.py

```diff
@@ -9,85 +10,181 @@
 
 class TestSpacyRecognizer(TestCase):
 
-    def test_person_first_name(self):
-        name = 'Dan'
-        results = spacy_recognizer.analyze(name, entities)
+# Test Name Entity 
+    # Bug #617 : Spacy Recognizer doesn't recognize Mr. Tailor as PERSON even though online spacy demo indicates that it does
+    # See http://textanalysisonline.com/spacy-named-entity-recognition-ner
+    # def test_person_first_name(self):
+    #     name = 'Dan'
+    #     results = spacy_recognizer.analyze(name, entities)
 
-        assert len(results) == 0
+    #     assert len(results) == 1
+    #     assert_result(results[0], entity[0], NER_STRENGTH)
 
     def test_person_first_name_with_context(self):
         name = 'Dan'
-        context = 'my name is '
-        results = spacy_recognizer.analyze(context + name, entities)
+        context= 'my name is'
+        results = spacy_recognizer.analyze('{} {}'.format(context, name), entities)
 
         assert len(results) == 1
-        assert results[0].score >= NER_STRENGTH
-        assert results[0].entity_type == entities[0]
-        assert results[0].start == 11
-        assert results[0].end == 14
+        assert_result_within_score_range(results[0], entities[0], 11, 14, NER_STRENGTH, 1)
 
     def test_person_full_name(self):
         name = 'Dan Tailor'
         results = spacy_recognizer.analyze(name, entities)
 
         assert len(results) == 1
-        assert results[0].score >= NER_STRENGTH
-        assert results[0].entity_type == entities[0]
-        assert results[0].start == 0
-        assert results[0].end == 10
+        assert_result(results[0], entities[0], 0, 10, NER_STRENGTH)
 
     def test_person_full_name_with_context(self):
         name = 'John Oliver'
-        results = spacy_recognizer.analyze(name + " is the funniest comedian", entities)
+        context = ' is the funniest comedian'
+        results = spacy_recognizer.analyze('{} {}'.format(name, context), entities)
 
         assert len(results) == 1
-        assert results[0].score >= NER_STRENGTH
-        assert results[0].entity_type == entities[0]
-        assert results[0].start == 0
-        assert results[0].end == 11
+        assert_result_within_score_range(results[0], entities[0], 0, 11, NER_STRENGTH, 1)
 
     def test_person_last_name(self):
         name = 'Tailor'
         results = spacy_recognizer.analyze(name, entities)
 
         assert len(results) == 0
+    
+    # Bug #617 : Spacy Recognizer doesn't recognize Mr. Tailor as PERSON even though online spacy visualizer indicates that it does
```

Comment:
Do we use the same version as they do? Anyway we'll probably replace Spacy soon. Pretty far from SOTA.

Reply:
I don't know which version they are using. We'll have to investigate when solving this bug in a separate PR.